### PR TITLE
Fixed several occurrences of inconsistent code.

### PR DESCRIPTION
**Pull request by martinschaef**
Fixed inconsistent code detected by our static analysis tool http://www.csl.sri.com/bixie-ws/

### CODE REVIEW

**Comment by ctomc** [comment]
this is ok to test

File: connector/src/main/java/org/jboss/as/connector/subsystems/datasources/AbstractDataSourceService.java [hunk 1]
```diff
@@ -187,22 +187,19 @@ protected synchronized void stopService() {
...
+                for (org.jboss.jca.core.api.management.DataSource mgtDs : deploymentMD.getDataSources()) {
+                    managementRepositoryValue.getValue().getDataSources().remove(mgtDs);
+                }
             }
-        }
-
-        if (deploymentMD.getConnectionManagers() != null) {
-            for (ConnectionManager cm : deploymentMD.getConnectionManagers()) {
-                cm.shutdown();
+
```
Comment (scottmarlow) [inline connector/src/main/java/org/jboss/as/connector/subsystems/datasources/AbstractDataSourceService.java]: You should see the same build errors locally if you do a "./build.sh clean install" from the root WildFly folder.  Seems to be a check-style complaint about line 195:
AbstractDataSourceService.java:195: Line has trailing spaces.

Comment (martinschaef) [inline connector/src/main/java/org/jboss/as/connector/subsystems/datasources/AbstractDataSourceService.java]: Changed it. Sorry, generated the patch from within github and didn't run checkstyle.

File: jpa/src/main/java/org/jboss/as/jpa/processor/PersistenceUnitServiceHandler.java [hunk 2]
```diff
@@ -171,12 +171,14 @@ private static void handleWarDeployment(DeploymentPhaseContext phaseContext, boo
...
             ArrayList<PersistenceUnitMetadataHolder> puList = new ArrayList<PersistenceUnitMetadataHolder>(1);
 
+            String deploymentRootName = null;
             // handle persistence.xml definition in the root of the war
             if (deploymentRoot != null &&
                 (holder = deploymentRoot.getAttachment(PersistenceUnitMetadataHolder.PERSISTENCE_UNITS)) != null &&
                 holder.getPersistenceUnits().size() > 0) {
                 // assemble and install the PU service
                 puList.add(holder);
+                deploymentRootName = deploymentRoot.getRootName()
```
Comment (scottmarlow) [inline jpa/src/main/java/org/jboss/as/jpa/processor/PersistenceUnitServiceHandler.java]: looks like a missing semicolon here is causing the next build error.  I didn't try to build your change either.

Comment (martinschaef) [inline jpa/src/main/java/org/jboss/as/jpa/processor/PersistenceUnitServiceHandler.java]: Uhhh ... Mondays are always hard. Sorry about that.

**Comment by bstansberry** [comment]
Thanks for this. It might take a while to get in as the PR touches on a bunch of different bits of code.

**Comment by martinschaef** [comment]
No problem. For me, it is more about checking if our static analysis makes sense. Btw., I'm not entirely sure if the patch for InMemoryDirectoryServiceFactory.java makes sense, because the procedure declares that it may throw something of type Exception, which would also cover RuntimeException. So maybe, this part of the patch is bogus.

**Comment by scottmarlow** [comment]
+1 for the JPA part of this change.

**Comment by kwart** [comment]
+1 for the `InMemoryDirectoryServiceFactory.java`

**Comment by bstansberry** [comment]
I've opened #7778 which is a rebased version of this. I also opened https://github.com/wildfly/wildfly-core/pull/894 which is for the one file here that has been moved to WildFly Core.

Thanks for this! All the issues your tool identified were valid.

**Pull request closed without merging**